DashSync/shared/Models/Managers/Chain Managers/DSMasternodeManager.m

```diff
@@ -501,14 +505,21 @@ - (void)processMasternodeListDiffResult:(DSMnDiffProcessingResult *)result forPe
         DSLog(@"•••• processMasternodeListDiffResult: missingMasternodeLists: %@", [self logListSet:neededMissingMasternodeLists]);
         UInt256 masternodeListBlockHash = masternodeList.blockHash;
         NSData *masternodeListBlockHashData = uint256_data(masternodeListBlockHash);
-        if ([neededMissingMasternodeLists count] && [self.store.masternodeListQueriesNeedingQuorumsValidated containsObject:masternodeListBlockHashData]) {
+        BOOL hasAwaitingQuorumValidation;
+        @synchronized (self.store.masternodeListQueriesNeedingQuorumsValidated) {
```

Comment:
@pankcuf I don't think @synchronized is a real solution, I personally against using @synchronized because it can lead performance decreasing and other "unknowns"